ecs/src/nox/ecs/EntityManager.cpp

```diff
@@ -33,6 +36,196 @@ nox::ecs::EntityManager::registerComponent(const MetaInformation& info)
+    {
+        for (const auto& type : list.second)
+        {
+            auto itr = std::find_if(std::begin(componentAccessLists),
+                                    std::end(componentAccessLists),
+                                    [type](const auto& listPair)
+                                    { return listPair.first == type; });
+            itr->second.insert(list.first);
+        }
+    }
+
+    std::vector<std::vector<TypeIdentifier>> executionOrder; 
+ 
+    while (!componentAccessLists.empty())
+    {     
+        //Sort so the list with largest amount of connections to others comes first
+        std::sort(std::begin(componentAccessLists),
+                  std::end(componentAccessLists),
+                  [](const auto& lhs, const auto& rhs)
+                  { return lhs.second.size() > rhs.second.size(); });
+
+        //Start of with the component type with the most connections
+        TypeIdentifier initialComponent = componentAccessLists.front().first;
+
+        executionOrder.emplace_back();
+        executionOrder.back().push_back(initialComponent);
+
+        TypeIdentifierSet connectedComponents(std::begin(componentAccessLists.front().second),
+                                              std::end(componentAccessLists.front().second),
+                                              TypeIdentifierComp()); //typeIdentifierComp);
+
+        //Need a copy to work with so deleting is not affecting the original
+        auto accessListsCopy = componentAccessLists;
+
+        //Done with the starting component, so erase it from the list
+        accessListsCopy.erase(std::begin(accessListsCopy));
+
+        while (!accessListsCopy.empty())
+        {
+            //Remove all component types that has a connection to component types already put in the executionOrder
+            for (auto itr = std::begin(accessListsCopy); itr != std::end(accessListsCopy);)
+            {
+                std::vector<TypeIdentifier> intersection;
+                std::set_intersection(std::begin(itr->second), std::end(itr->second),
+                                      std::begin(executionOrder.back()), std::end(executionOrder.back()),
+                                      std::back_inserter(intersection),
+                                      typeIdentifierComp);
+                
+                if (!intersection.empty())
+                {
+                    itr = accessListsCopy.erase(itr);
+                }
+                else
+                {
+                    ++itr;
+                }
+            }
+
+            //Ensure not to try to add another component to the executionOrder when there are none left
+            if (accessListsCopy.empty())
+            {
+                break;
+            }
+
+            //Lambda less than comparison between which component types has more relations to the connectedComponents
+            auto compareFunction = [&connectedComponents, typeIdentifierComp](auto& lhs, auto& rhs)
```

Comment:
const auto&, you are not supposed to change anything, however that indication is given.